Coding/Java/src/org/jmol/adapters/readers/quantum/QCJSONReader.java

```diff
@@ -0,0 +1,377 @@
+      if (isFractional) {
+        f = QCSchemaUnits.getConversionFactor(atoms, "unit_cell", QCSchemaUnits.UNITS_ANGSTROMS);
+        double[] cell = QCSchemaUnits.getDoubleArray(atoms, "unit_cell");
+        // a b c alpha beta gamma 
+        // m.m00, m.m10, m.m20, // Va
+        // m.m01, m.m11, m.m21, // Vb
+        // m.m02, m.m12, m.m22, // Vc
+        // dimension, (float) volume,
+        if (cell == null) {
+          Logger.error("topology.unit_cell is missing even though atoms are listed as fractional");
+        } else {
+          for (int i = 0; i < 6; i++) {
+            switch (i) {
+            case 3:
+              f = 1;
+              //$FALL-THROUGH$
+            default:
+              setUnitCellItem(i, (float)(cell[i] * f));
+              break;
+            }
+          }
+        }
+      }
+      for (int i = 0, pt = 0; i < modelAtomCount; i++) {
+        Atom atom = asc.addNewAtom();
+        setAtomCoordXYZ(atom, (float)(coords[pt++] * f), (float)(coords[pt++] * f), (float) (coords[pt++]
+            * f));
+        String sym = (symbols == null ? JmolAdapter
+            .getElementSymbol(atomNumbers[i]) : symbols[i]);
+        atom.atomName = (atom_names == null ? sym : atom_names[i]);
+        atom.elementNumber = (short) (atomNumbers == null ? JmolAdapter
+            .getElementNumber(sym) : atomNumbers[i]);
+      }
+      if (doReadMolecularOrbitals) {
+        readMolecularOrbitals(getMapSafely(step, "molecular_orbitals"));
+        clearOrbitals();
+      }
+      applySymmetryAndSetTrajectory();
+      if (loadVibrations) {
+        readFreqsAndModes(QCSchemaUnits.getList(step, "vibrations"));
+      }
+      
+    }
+  }
+  
+  private boolean readFreqsAndModes(ArrayList<Object> vibrations) throws Exception {
+    //  "frequency":{"value":-0.00,"units":["cm^-1","?"]},
+    //      "ir_intensity":{"value":0.000005,"units":["au",1]},
+    //    "vectors":[
+
+    if (vibrations != null) {
+      int n = vibrations.size();
+      for (int i = 0; i < n; i++) {
+        @SuppressWarnings("unchecked")
+        Map<String, Object> vib = (Map<String, Object>) vibrations.get(i);
+        double freq = QCSchemaUnits.getDouble(vib, "frequency", QCSchemaUnits.UNITS_CM_1);
+        double[] vectors = QCSchemaUnits.getDoubleArray(vib, "vectors");
+        if (i > 0)
+          asc.cloneLastAtomSet();
+        asc.setAtomSetFrequency(null, null, "" + freq, QCSchemaUnits.UNITS_CM_1);
+        int i0 = asc.getLastAtomSetAtomIndex();
+        for (int j = 0, pt = 0; j < modelAtomCount; j++) {
+          asc.addVibrationVector(j + i0, (float) (vectors[pt++] * ANGSTROMS_PER_BOHR),
+              (float) (vectors[pt++] * ANGSTROMS_PER_BOHR), (float) (vectors[pt++]
+                  * ANGSTROMS_PER_BOHR));
+        }
+      }
+    }
+    return true;
+  }
+  
+  private boolean haveEnergy = true;
+  
+  /**
+   * Read basis and orbital information.
+   * 
+   * @param molecular_orbitals
```

Comment:
Can we provide much more information of what the `molecular_orbitals` object entails?